### ISSUE TITLE
Allow access to restricted content using YouTube login cookie. 

### ISSUE THREAD

**Issue by funsafe-ptr** [MaintainTeam/LastPipeBender#13]
### Checklist

- [X] I made sure that there are *no existing issues* - [open](https://github.com/polymorphicshade/Tubular/issues) or [closed](https://github.com/polymorphicshade/Tubular/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [X] I have read the [FAQ](https://newpipe.net/FAQ/) and my problem isn't listed.
- [X] I'm aware that this is a request for Tubular itself and that requests for adding a new service need to be made in [vanilla NewPipeExtractor](https://github.com/TeamNewPipe/NewPipeExtractor/issues).
- [X] I have taken the time to fill in all the required details. I understand that the feature request will be dismissed otherwise.
- [X] This issue contains only one feature request.
- [X] I have read and understood the vanilla NewPipe [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/dev/.github/CONTRIBUTING.md).

### Feature description

Allow access to restricted content using YouTube login cookie. And use the cookie only if required.

### Why do you want this feature?

YouTube now requires an account to access restricted content.

### Why ist the feature relevant to this fork?

Upstream issue marked "wontfix".
https://github.com/TeamNewPipe/NewPipe/issues/11382

### Additional information

https://codeberg.org/NullPointerException/PipePipeClient/src/branch/dev/app/src/main/java/org/schabi/newpipe/settings/YouTubeAccountSettingsFragment.java
https://codeberg.org/NullPointerException/PipePipeClient/src/branch/dev/app/src/main/java/org/schabi/newpipe/views/YouTubeLoginWebViewActivity.java
https://codeberg.org/NullPointerException/PipePipeExtractor

Like you said, PipePipe is a hard fork and might be tough to merge. 😓

Anyway, I don’t know much Java, and I’m using my phone 📱, so I’m not sure how much I can help. 😕

**Comment by funsafe-ptr** [comment]
Yes, I think? I can play restricted content 🔒 until the end, for more than 1 minute. Normal content shows a 403 error as usual after more than 1 minute. I'm using downgraded PipePipe 4.0.2.

**Comment by asandikci** [comment]
Are trying normal content with or without account?

**Comment by funsafe-ptr** [comment]
I think PipePipe 4.0.2 is using yt-dlp only as a fallback? 🤔 It loads fast for normal content, just like the usual NewPipe 🚀, but it's slow to load 🐢 for restricted content, similar to normal content in the latest version.

**Comment by funsafe-ptr** [comment]
Now, even with a VPN, after a few hours, it requires me to log in to prove I'm not a bot 😩

**Comment by jimbattun** [comment]
> Could anyone try the YT login in PipePipe (previous versions than 4.1.0) to see if it solves IP ban / 403 problem? (#12)

I tested PipePipe 4.1.0, and without signing in to YouTube, I encountered the error: "Your IP has been temporarily blocked by YouTube. For a period of time, you will experience issues when playing certain videos." After signing in via Settings → Accounts → YouTube, the videos started playing normally. The only issue I noticed was the loading speed—after clicking on a video, playback starts after approximately 10 seconds.
